partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt

```diff
@@ -637,6 +638,11 @@ internal object RexConverter {
             if (id.matches("EXISTS", ignoreCase = true)) {
                 return visitExprCallExists(node, context)
             }
+
+            if (isDateAddFunction(id)) {
+                return visitExprCallDateAddFunction(id, node, context)
+            }
+
```

Comment:
This is still an issue from the prior review. You're performing the validation that this is a `date_add_<datetime field>` function and then performing it again in `visitExprCallDateAddFunction` with different logic. We really should just be using **one** validation that also extracts out the datetime field.